src/Integration.Vsix/CFamily/CLangAnalyzer.cs

```diff
@@ -73,66 +71,85 @@ public bool IsAnalysisSupported(IEnumerable<AnalysisLanguage> languages)
             return languages.Contains(AnalysisLanguage.CFamily);
         }
 
-        public void ExecuteAnalysis(string path, string charset, IEnumerable<AnalysisLanguage> detectedLanguages,
-            IIssueConsumer consumer, IAnalyzerOptions analyzerOptions,
-            CancellationToken cancellationToken)
+        public void ExecutePreAnalysis(string path, IEnumerable<AnalysisLanguage> detectedLanguages, CFamilyAnalyzerOptions analyzerOptions, CancellationToken cancellationToken)
         {
-            ExecuteAnalysis(path, detectedLanguages, consumer, analyzerOptions, analysisStatusNotifier, cancellationToken);
-        }
+            var request = CreateRequest(path, detectedLanguages, analyzerOptions, null);
 
-        public void ExecuteAnalysis(string path,
-            IEnumerable<AnalysisLanguage> detectedLanguages,
-            IIssueConsumer consumer,
-            IAnalyzerOptions analyzerOptions,
-            IAnalysisStatusNotifier statusNotifier,
-            CancellationToken cancellationToken)
-        {
-            var projectItem = dte?.Solution?.FindProjectItem(path);
-            if (projectItem == null)
+            if (request == null)
             {
                 return;
             }
 
-            Debug.Assert(IsAnalysisSupported(detectedLanguages));
+            TriggerPreAnalysisAsync(request, cancellationToken).Forget();
+        }
+
+        public void ExecuteAnalysis(string path, string charset, IEnumerable<AnalysisLanguage> detectedLanguages, IIssueConsumer consumer, IAnalyzerOptions analyzerOptions, CancellationToken cancellationToken)
+        {
+            var request = CreateRequest(path, detectedLanguages, analyzerOptions, cFamilyRulesConfigProvider);
 
-            var request = CreateRequest(logger, projectItem, path, cFamilyRulesConfigProvider, analyzerOptions);
             if (request == null)
             {
                 return;
             }
+            
+            TriggerAnalysis(request, consumer, cancellationToken);
+        }
+
+        private Request CreateRequest(string path, IEnumerable<AnalysisLanguage> detectedLanguages, IAnalyzerOptions analyzerOptions, ICFamilyRulesConfigProvider rulesConfigProvider)
```

Comment:
Helper method for repeating code. It accepts rulesConfigProvider that can be null now when called from PreAnalysis